alert-database/src/main/java/com/synopsys/integration/alert/database/api/DefaultJobAccessor.java

```diff
@@ -70,6 +71,28 @@ public DefaultJobAccessor(ConfigGroupRepository configGroupRepository, DefaultCo
         return convertToJobModels(jobEntities);
     }
 
+    @Override
+    public List<ConfigurationJobModel> getMatchingEnabledJobs(FrequencyType frequency, Long providerConfigId, NotificationType notificationType) {
+        //TODO change this to return a page of results
+        List<UUID> matchingJobIds = configGroupRepository.findMatchingEnabledJobIds(frequency.name(), String.valueOf(providerConfigId), notificationType.name());
+        if (matchingJobIds.isEmpty()) {
+            return List.of();
+        }
+        List<ConfigGroupEntity> jobEntities = configGroupRepository.findByJobIds(matchingJobIds);
+        return convertToJobModels(jobEntities);
+    }
+
+    @Override
+    public List<ConfigurationJobModel> getMatchingEnabledJobs(Long providerConfigId, NotificationType notificationType) {
+        //TODO change this to return a page of results
+        List<UUID> matchingJobIds = configGroupRepository.findMatchingEnabledJobIds(String.valueOf(providerConfigId), notificationType.name());
+        if (matchingJobIds.isEmpty()) {
+            return List.of();
+        }
+        List<ConfigGroupEntity> jobEntities = configGroupRepository.findByJobIds(matchingJobIds);
+        return convertToJobModels(jobEntities);
+    }
```

Comment:
I believe these can also use a common private method that is passed a bifunction. You would pass the findMatchingEnabledJobIds function in.